src/main/java/org/broadinstitute/hellbender/utils/fermi/FermiLiteAssembly.java

```diff
@@ -17,6 +18,25 @@ public FermiLiteAssembly( final List<Contig> contigs ) {
     public Contig getContig( final int idx ) { return contigs.get(idx); }
     public List<Contig> getContigs() { return contigs; }
 
+    public int computeN50() {
+        final int nContigs = contigs.size();
+        if ( nContigs < 1 ) return 0;
+        if ( nContigs == 1 ) return contigs.get(0).getSequence().length;
+
+        final int[] lengths = new int[nContigs];
+        for ( int idx = 0; idx != nContigs; ++idx ) {
+            lengths[idx] = contigs.get(idx).getSequence().length;
+        }
+        Arrays.sort(lengths);
+        final int totalSize = Arrays.stream(lengths).sum();
+        int runningTotal = 0;
+        for ( int idx = nContigs - 1; idx >= 0; --idx ) {
+            runningTotal += 2 * lengths[idx];
+            if ( runningTotal >= totalSize ) return lengths[idx];
```

Comment:
Total nitpick, but according to the definition I found, N50 is the longest contig length such that 50% or more of the genome is in a contig that large or greater. In which case this should check "runningTotal > totalSize" (counter-example, lengths = {4, 4, 8} should yield 8).

Reply:
Hmm.  Looks to me like it's correct.  Did you notice that the indices are running backwards?  (I'll add a comment and some javadoc.)
For your example:
totalSize = 4+4+8 = 16
first time through loop,
idx = 2
lengths[2] = 8
runningTotal = 2 * 8 = 16
runningTotal >= totalSize, so return lengths[2] = 8

Reply:
I did *NOT* notice that the indices were running backwards. My mistake, your code is correct.